rust/server/src/vss_service.rs

```diff
@@ -0,0 +1,154 @@
+use http_body_util::{BodyExt, Full};
+use hyper::body::{Bytes, Incoming};
+use hyper::service::Service;
+use hyper::{Error, Request, Response, StatusCode};
+use std::collections::HashMap;
+
+use prost::Message;
+
+use api::auth::Authorizer;
+use api::error::VssError;
+use api::kv_store::KvStore;
+use api::types::{
+	DeleteObjectRequest, DeleteObjectResponse, ErrorCode, ErrorResponse, GetObjectRequest,
+	GetObjectResponse, ListKeyVersionsRequest, ListKeyVersionsResponse, PutObjectRequest,
+	PutObjectResponse,
+};
+use std::future::Future;
+use std::pin::Pin;
+use std::sync::Arc;
+
+#[derive(Clone)]
+pub struct VssService {
+	store: Arc<dyn KvStore>,
+	authorizer: Arc<dyn Authorizer>,
```

Comment:
Why not use type parameters?

Reply:
Sure, in `main.rs` it could be parameterized that way, but that doesn't require having `dyn` leak into the library.

Reply:
There's a runtime cost for `dyn` as it results in dynamic dispatch and prevents the compiler from inlining. The tradeoff is larger binary size as the code will be duplicated for each instantiation.

https://doc.rust-lang.org/std/keyword.dyn.html#trade-offs

Reply:
Hmm... I guess in LDK Node we are supporting that but just hiding `dyn` in a type alias.

https://github.com/lightningdevkit/ldk-node/blob/aaccde78ce072d2eed345c7c0cc23006eb0d4d2a/src/types.rs#L34

So maybe fine to just do the same.